src/SFML/Window/Win32/WindowImplWin32.cpp

```diff
@@ -56,67 +56,72 @@
 
 namespace
 {
+constexpr int maxTouchPoints = 10; // Current Windows API limitation
+
+// Define constants for tablet API to avoid pulling in an extra Tablet PC header
+const char*              tabletServiceProperty     = "MicrosoftTabletPenServiceProperty";
+const unsigned long long tabletDisablePressAndHold = 0x00000001ULL;
+
 unsigned int               windowCount      = 0; // Windows owned by SFML
 unsigned int               handleCount      = 0; // All window handles
 const wchar_t*             className        = L"SFML_Window";
 sf::priv::WindowImplWin32* fullscreenWindow = nullptr;
+DWORD                      touchIDs[maxTouchPoints];
+POINT                      touches[maxTouchPoints];
 
-const GUID GUID_DEVINTERFACE_HID = {0x4d1e55b2, 0xf16f, 0x11cf, {0x88, 0xcb, 0x00, 0x11, 0x11, 0x00, 0x00, 0x30}};
-
-void setProcessDpiAware()
+// Convert a hardware dependent ID to a 0 based index we can use
+int getTouchID(DWORD id)
 {
-    // Try SetProcessDpiAwareness first
-    HINSTANCE shCoreDll = LoadLibrary(L"Shcore.dll");
-
-    if (shCoreDll)
+    for (int i = 0; i < maxTouchPoints; ++i)
     {
-        enum ProcessDpiAwareness
-        {
-            ProcessDpiUnaware         = 0,
-            ProcessSystemDpiAware     = 1,
-            ProcessPerMonitorDpiAware = 2
-        };
-
-        using SetProcessDpiAwarenessFuncType = HRESULT(WINAPI*)(ProcessDpiAwareness);
-        auto SetProcessDpiAwarenessFunc      = reinterpret_cast<SetProcessDpiAwarenessFuncType>(
-            reinterpret_cast<void*>(GetProcAddress(shCoreDll, "SetProcessDpiAwareness")));
-
-        if (SetProcessDpiAwarenessFunc)
+        if (touchIDs[i] == id)
+            return i;
+    }
+    for (int i = 0; i < maxTouchPoints; ++i)
+    {
+        if (touchIDs[i] == static_cast<DWORD>(-1))
```

Comment:
Casting a `-1` to an unsigned int seems odd. Do we really intend to do an underflow?

Reply:
There's no strict definition on whether 0 for example would be a regular/valid value; it's hardware/driver dependent apparently. Just using `std::optional` as suggested below might be the better approach here now that's possible.